tests/sources/test_redis.py

```diff
@@ -213,3 +218,105 @@ async def test_get_db_records():
+
+
+@pytest.mark.parametrize(
+    "filtering",
+    [
+        Filter(
+            {
+                ADVANCED_SNIPPET: {
+                    "value": [
+                        {"database": 0},
+                    ]
+                }
+            }
+        ),
+    ],
+)
+@pytest.mark.asyncio
+@freeze_time("2023-01-24T04:07:19+00:00")
+async def test_get_docs_with_sync_rules(filtering):
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.get_db_records = AsyncIterator(items=DOCUMENT)
+        async for (doc, _) in source.get_docs(filtering):
+            assert doc in DOCUMENT
+
+
+@pytest.mark.parametrize(
+    "advanced_rules, expected_validation_result",
+    [
+        (
+            # valid: empty array should be valid
+            [],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: empty object should also be valid -> default value in Kibana
+            {},
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: one custom pattern
+            [{"database": 0, "key_pattern": "*"}],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: two custom patterns
+            [
+                {"database": 0, "key_pattern": "test*"},
+                {"database": 1, "_type": "string"},
+            ],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # invalid: database number
+            [{"database": -1}],
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
+        (
+            # invalid: array of arrays -> wrong type
+            {"database": ["a/b/c", ""]},
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_advanced_rules_validation(advanced_rules, expected_validation_result):
+    async with create_redis_source() as source:
+        source.client._client = RedisClientMock()
+        validation_result = await RedisAdvancedRulesValidator(source).validate(
+            advanced_rules
+        )
+        assert validation_result == expected_validation_result
+
+
+@pytest.mark.asyncio
+async def test_advanced_rules_validation_for_invalid_db():
+    async with create_redis_source() as source:
+        mocked_client = Mock()
+        mocked_client.validate_database = AsyncMock(return_value=True)
```

Comment:
Don't see why this test case pass. `validate_database` always returns true.

Can this be added as one of the test cases of the above test?